pkg/ui/src/views/reports/containers/debug/index.tsx

```diff
@@ -79,285 +80,399 @@ function DebugPanelLink(props: { name: string, url: string,  note: string }) {
+}
+
+interface IDebugState {
+  selected: string;
+  location: string;
+  opened: boolean;
+  width: number;
+}
+
+const pattern = /[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@%_\+.~#?&=]*)?/;
+// tslint:disable-next-line: variable-name
+class Debug extends React.Component<IDebugProps, IDebugState> {
+  state = {
+    selected: "all",
+    location: "",
+    opened: false,
+    width: window.innerWidth,
+  };
+
+  private rangeContainer = React.createRef<HTMLDivElement>();
+
+  componentDidMount() {
+    this.setState({
+      selected: window.location.hostname,
+    });
+    window.addEventListener("resize", this.updateDimensions);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener("resize", this.updateDimensions);
+  }
+
+  getOptionsForSelect = () => this.props.nodesSummary.nodeIDs.map((value: string) => ({
+    value: this.props.nodesSummary.nodeDisplayNameByID[value],
+    label: `Node ${value}`,
+  }))
 
-        <div className="debug-header__annotations">
-          <LicenseType />
-          <DebugAnnotation label="Web server" value={ `n${NODE_ID}` } />
+  openApp = () => {
+    const { location } = this.state;
+    window.open(`http://${location}/#/debug/`, "_self");
+  }
+
+  selectApp = (app: DropdownOption) => () => this.setState({ selected: app.value, location: app.value.match(pattern)[0] }, this.openApp);
+
+  toggleDropDown = () => this.setState({ opened: !this.state.opened });
+
+  arrowRenderer = (isOpen: boolean) => {
+    if (isOpen) {
+      return <span><Icon type="caret-up" /></span>;
+    }
+    return <span className="active"><Icon type="caret-down" /></span>;
+  }
+
+  renderOptions = () => {
+    const { selected } = this.state;
+    return this.getOptionsForSelect().map((option: DropdownOption) => (
+      <a onClick={this.selectApp(option)} className={`option ${selected === option.value ? "selected" : ""}`}>{option.label}</a>
+    ));
+  }
+
+  updateDimensions = () => {
+    this.setState({
+      width: window.innerWidth,
+    });
+  }
+
+  render() {
+    const { selected, width, opened } = this.state;
+    const containerLeft = this.rangeContainer.current ? this.rangeContainer.current.getBoundingClientRect().left : 0;
+    const left = width >= (containerLeft + 150) ? -45 : width - (containerLeft + 150);
+    return (
+      <div className="section">
+        <Helmet title="Debug" />
+        <h1 className="base-heading">Advanced Debugging</h1>
+        <div className="debug-header">
+          <InfoBox>
+            <p>
+              The following pages are meant for advanced monitoring and troubleshooting.
+              Note that these pages are experimental and undocumented. If you find an issue,
+              let us know through{" "}
+              <a href={ COMMUNITY_URL }>these channels.</a>
+            </p>
+          </InfoBox>
+          <div className="debug-header__annotations">
+            <LicenseType />
+            <Dropdown
```

Comment:
This file is already very long, please extract this into a separate component in its own file.